pkg/cmd/pulumi/plugin_install.go

```diff
@@ -103,6 +103,18 @@ func newPluginInstallCmd() *cobra.Command {
 					Checksums:         checksums,
 				}
 
+				// bundled plugins are not installable with this command.
+				// They are expected to be distributed with Pulumi itself.
+				if workspace.IsBundledPlugin(pluginSpec.Kind, pluginSpec.Name) {
+					return fmt.Errorf(
+						"the %v %v plugin is bundled with Pulumi, and cannot be directly installed"+
+							" with this command. If you need to reinstall this plugin, uninstall"+
+							" Pulumi via your package manager and re-install.",
+						pluginSpec.Name,
+						pluginSpec.Kind,
+					)
+				}
+
```

Comment:
Can you add a small test on this to ensure that this check is run?

Reply:
You'll want to follow @abhinav's style of splitting command code out of the cobra callback for that.

Reply:
How would you want the control flow for the test?
Because this is in the top-level of the cobra cmd, we'd have to mock out the function, add a faker interface, and check the fake. Currently, this function has no tests at all.

Reply:
@Frassle link?

Reply:
eg https://github.com/pulumi/pulumi/pull/12499